tls/s2n_prf.c

```diff
@@ -196,39 +200,41 @@ static int s2n_evp_hmac_p_hash_free(union s2n_prf_working_space *ws)
     return 0;
 }
 
-static int s2n_hmac_p_hash_new(union s2n_prf_working_space *ws)
+static int s2n_hmac_p_hash_new(struct s2n_prf_working_space *ws)
 {
-    return 0;
+    GUARD(s2n_hmac_new(&ws->tls.p_hash.s2n_hmac));
+
+    return s2n_hmac_init(&ws->tls.p_hash.s2n_hmac, S2N_HMAC_NONE, NULL, 0);
```

Comment:
s2n_hmac_p_hash_new is called via s2n_prf_new() when out of FIPS mode and thus, is using s2n's HMAC implementation for the p_hash. 

Currently, if s2n is not in FIPS mode, s2n_hmac_new uses the low_level s2n_hash and this is a no-op. If we change our logic such that s2n_hash always uses the high_level EVP's, this will not be a no-op. So, it's just ensuring that we are using the s2n_hmac API's in a consistent manor to prevent future changes from breaking us.